assets/js/atomic/blocks/product-elements/rating/edit.tsx

```diff
@@ -3,22 +3,41 @@
  */
 import { __ } from '@wordpress/i18n';
 import { useBlockProps } from '@wordpress/block-editor';
+import type { ProductQueryContext as Context } from '@woocommerce/type-defs/contexts';
+import { useEffect } from 'react';
 
 /**
  * Internal dependencies
  */
 import Block from './block';
 import withProductSelector from '../shared/with-product-selector';
 import { BLOCK_TITLE, BLOCK_ICON } from './constants';
+import { Attributes } from './types';
 
-const Edit = ( { attributes } ) => {
+interface Props {
+	attributes: Attributes;
+	setAttributes: ( attributes: Record< string, unknown > ) => void;
```

Comment:
🤔  Shouldn't `attributes` be of the type `Attributes` above?

It seems to me that these interface might be lacking all the other props, however… you can use the handy `BlockEditProps< T >` already available in the types.

Reply:
Addressed in ad2b4decd.